File: di_java/src/test/java/com/kmmania/mas/DAOImplMASTest.java
package com.kmmania.mas;

import com.kmmania.api.DAO;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.*;

class DAOImplMASTest {
    @InjectMocks
    private DAO daoMock;

    @BeforeEach
    void setUp() {
        daoMock = mock(DAOImplMAS.class);
    }

    @Test
    @DisplayName("getData() method should retrieve data from DAO")
    void getDataShouldRetrieveDataFromDAO() {
        // Given
        when(daoMock.getData()).thenReturn(17.5);

        // When
        double result = daoMock.getData();

        // Then
        verify(daoMock, times(1)).getData();
        assertThat(result).isEqualTo(17.5);
    }
}

File: di_kotlin/src/test/kotlin/mas/DAOImplMASTest.kt
package mas

import com.kmmania.api.DAO
import com.kmmania.mas.DAOImplMAS
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.InjectMocks
import org.mockito.Mockito.*

class DAOImplMASTest {
    @InjectMocks
    private lateinit var daoMock: DAO

    @BeforeEach
    fun setUp() {
        daoMock = mock(DAOImplMAS::class.java)
    }

    @Test
    fun `getData() should retrieve data from DAO`() {
        // Given
        `when`(daoMock.getData()).thenReturn(17.5)

        // When
        val result = daoMock.getData()

        // Then
        verify(daoMock, times(1)).getData()
        assertThat(result).isEqualTo(17.5)
    }
}